### PR TITLE
HP-1410: bank details

### PR DESCRIPTION
**Pull request by tafid**

### CODE REVIEW

File: src/widgets/BankDetailsFormWidget.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+class BankDetailsFormWidget extends Widget
+{
+    public Requisite|Contact $parentModel;
+    public ActiveForm $form;
+    public CrudController $controller;
+
+    public function run()
+    {
+        $currencies = array_keys($this->controller->getCurrencyTypes());
```
Comment (SilverFire) [inline src/widgets/BankDetailsFormWidget.php:21]: I think we should render this widget only for contacts, that represent requisites. @hiqsol what do you think?

File: src/models/BankDetails.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    public function attributeLabels()
+    {
+        return [
+            'bank_name' => Yii::t('hipanel:client', 'Bank name'),
+            'bank_account' => Yii::t('hipanel:client', 'Bank account'),
+            'bank_address' => Yii::t('hipanel:client', 'Bank address'),
+            'bank_swift' => Yii::t('hipanel:client', 'SWIFT code'),
+            'bank_correspondent' => Yii::t('hipanel:client', 'Correspondent bank'),
+            'bank_correspondent_swift' => Yii::t('hipanel:client', 'Correspondent bank SWIFT code'),
```
Comment (SilverFire) [inline src/models/BankDetails.php:53]: Could you add translations, please?

**Review by SilverFire — CHANGES_REQUESTED** [review]

**Comment by SilverFire** [comment]
It seems that a RequisitesGridView is no longer used after this pull request. Maybe, we should drop it?

File: src/widgets/BankDetailsSummaryRenderWidget.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return implode("\n\n", $result);
+    }
+
+    private function renderBankAccount(BankDetails $model): ?string
+    {
+        if (empty($model->bank_account)) {
+            return null;
+        }
+        $currency = '';
+//        $currency = mb_strtoupper($model->currency);  // todo: adjust currency display
```
Comment (SilverFire) [inline src/widgets/BankDetailsSummaryRenderWidget.php:38]: Currency seems to be always an empty string

**Review by SilverFire — CHANGES_REQUESTED** [review]

File: src/widgets/BankDetailsSummaryRenderWidget.php (hunk 3 above)
Comment (tafid) [reply src/widgets/BankDetailsSummaryRenderWidget.php:38]: so far it's meant to be

**Review by SilverFire — APPROVED** [review]

**Pull request merged**